[procps] New versions

greenc-FNAL opened the pull request:
- [procps] New versions 3.3.14, 3.3.3.15..3.3.17, 4.0.0..4.0.3

~Blocked by #38646.~

greenc-FNAL committed:
[procps] New versions 3.3.14, 3.3.3.15..3.3.17, 4.0.0..4.0.3

bebosudo commented on var/spack/repos/builtin/packages/procps/package.py:
```diff
@@ -27,22 +35,22 @@ class Procps(AutotoolsPackage):
+    # Need to tell the build to use the tools it already has to find
+    # libintl (if appropriate).
+    patch("libintl-3.3.14.patch", when="@3.3.14:3.3")
+    patch("libintl-4.0.0.patch", when="@=4.0.0")
```
```suggestion
    patch("libintl-4.0.0.patch", when="@4.0.0")
```

From a quick grep this seems to be the preferred syntax when a specific match is required?

greenc-FNAL replied:
Your suggested patch would match (e.g.) 4.0.0.3; my original matches 4.0.0 and nothing else. The `@=` notation is relatively new.

bebosudo replied:
Oh, I didn't know about that syntax. Thanks!

bebosudo reviewed (approved):
I tested this on my fedora box compiling all the versions added here, and the patches work great!

greenc-FNAL commented:
@spackbot help

scheibelp commented on var/spack/repos/builtin/packages/procps/libintl-4.0.1.patch:
```diff
@@ -0,0 +1,29 @@
+@@ -1,7 +1,6 @@
+ 
+-CYGWINFLAGS =
++CYGWINFLAGS = $(LTLIBINTL)
```
This looks the same as [libintl-4.0.0.patch](https://github.com/spack/spack/pull/38650/files#diff-01faf2a0353dd738aa873c4b2db144b7ac51fb7e3fcbea714abff58460fc823e) except for line numbers. I'm wondering if we can use a `patch()` function with `FileFilter` (generally that would be used for sed-style examine one line at a time, but it would be nice to be able to consolidate these changes since they are quite similar).

greenc-FNAL replied:
It's not the line numbers (`patch` is pretty good if only the line numbers differ), it's the context: `kill.1` -> `man/kill.1`, `kill_SOURCES` -> `src_kill_SOURCES`, etc.

We could use `FileFilter`, but I'm inclined to keep the patch format because of https://gitlab.com/procps-ng/procps/-/merge_requests/189, and the fact that there shouldn't be any more patches.

greenc-FNAL replied:
Note that https://gitlab.com/procps-ng/procps/-/merge_requests/189 was merged with minor changes and should be included in `procps` 4.0.4.

greenc-FNAL commented:
@scheibelp Bump?

tldahlgren reviewed (commented):
FYI. Confirmed all 8 of the new version sha256

greenc-FNAL commented:
> FYI. Confirmed all 8 of the new version sha256

Not sure what you mean by this. Is this something I should have done/checked?

greenc-FNAL committed:
Put upper bound on patch application

* [Patch was
  merged](https://gitlab.com/procps-ng/procps/-/merge_requests/189) for
  upcoming 4.0.4.

tldahlgren commented:
> > FYI. Confirmed all 8 of the new version sha256
> 
> Not sure what you mean by this. Is this something I should have done/checked?

You didn't do anything wrong.  It's something I like to check for PRs since erroneous ones sometimes slip in, especially when there are more than a few being added to a PR.
